packages/strapi/lib/middlewares/parser/index.js

```diff
@@ -16,7 +16,7 @@ const addQsParser = app => {
     get() {
       const qstr = this.querystring;
       const cache = (this._querycache = this._querycache || {});
-      return cache[qstr] || (cache[qstr] = qs.parse(qstr, { depth: 20 }));
+      return cache[qstr] || (cache[qstr] = qs.parse(qstr, { arrayLimit: 100, depth: 20 }));
```

Comment:
I think we should alow passing some options so we can have a sensible default and leave the rest to the user. 

We can achieve this by adding a `queryStringParser` to `strapi.config.middleware.settings.parser` and setting the default in the default.json and passing the option to the `addQsParser` fn. We need to make sure we do not pass the option to koa-body though.